### ISSUE TITLE
Engine.update() with a large delta crashes browser tab

### ISSUE THREAD

**Issue by anthonygood** [liabru/matter-js#816]
I inadvertently passed a big delta to `Engine.update` and ended up crashing my browser tab with 100% CPU usage.

This code reproduces it for me but I suppose the actual delta value might depend on the hardware.

```js
const crash = () => {
  const { Engine, Bodies, World } = Matter
  const engine = Engine.create()
  const body = Bodies.circle(0, 0, 1)

  World.add(engine.world, body)
  Engine.update(engine, 1000000)
}
```

This Codepen crashes for me in Safari, Chrome seems OK but with high CPU usage: https://codepen.io/scriptie/pen/MWYXWyW?editors=0011
![image](https://user-images.githubusercontent.com/3775267/72224742-ae4dc700-3575-11ea-8a6b-f3897cb78b48.png)

**Comment by wassfila** [comment]
@anthonygood please feel free to have a look at an example solution I linked to this ticket

**Comment by LeotomasMC** [comment]
well you see, its trying to run a lot of time in a little bit of time, its like if your playing a game, and you try to run it at 1000x speed, it doesnt work very well trying to do 1000x more things at the same time.

**Comment by anthonygood** [comment]
@LeotomasMC Didn't seem sarcastic! I assumed it was to do with the number of calculations involved (bigger delta == more stuff I guess).

I suppose from a usability perspective I'd enforce some kind of (configurable) delta cap to prevent locking the browser. I appreciate it's the library consumer's responsibility, but it would be a 'nice-to-have' to avoid having to kill browser tabs during development.

**Comment by LeotomasMC** [comment]
the delta is the delta time, or how much time it is trying calculate for. im not sure exactly how its implemented though (is it doing many updates, or 1 update calculating longer time, or a mix of both).

out of curiosity, how do you end up with a large enough delta to crash the browser?

**Comment by wassfila** [comment]
@anthonygood I renamed the const to max_cpu_ms in the [last version ](https://github.com/NetworkGraphs/graph2d/blob/master/src/physics_matter.js). Which is configurable to the lowest expected fps, here 10 fps for 100 ms. But can be configured down to 3 or 1 fps if needed. Note that in case the fps is tool low, there is no solution to prevent instabilities as the jitter becomes too big to get compensated by the delta corrections.
@LeotomasMC in case the tab is in the background, and the user reactivate it after 5 min or more, that delta time, if fed to the physics engine result in a huge number that can result in lots of NaN and might lead to crashing the browser I assume. So limiting the max delta time fed to the physics engine is important.

**Comment by liabru** [comment]
Yeah I'd say this is down to the library user to maintain a reasonable delta but I know what you mean. I think this is already implemented in the built in runner, but intentionally not directly in the engine which is considered lower level. 

Not sure what is going on specifically to cause this but essentially the lack of accuracy caused by a high delta causes growing instability, so even if any errors were fixed it would definitely still be unusable!
